### PR TITLE
#HaGhI3pb Authentication

### PR DESCRIPTION
**Pull request by Eli250**
### What does this PR do?

Authenticate Private End Points

### Description of Task to be completed?

The main task was to allow only authenticated user to manipulate data on private end points such as create, update and delete of articles, queries and users.

### Relevant pivotal tracker stories?

HaGhI3pb

### How can you manually test this
Clone this repository my-brand-api

After that create a .env file and include these lines in a file:
DEVELOPMENT_DB="url:port/dbname"
TEST_DB="url:port/dbname"
PRODUCTION_DB="url:port/dbname"

Replace "url:port/dbname" with you database url and port respectively.

After that run this command npm -y install to install all project dependencies and you are good to go
auth
auth2
.

### CODE REVIEW

File: src/controllers/userController.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        };
+        const createdUser = await createUser(user);
+        res.status(201).json({
+          status: 201,
+          message: "User Registered!",
+          user: createdUser,
+        });
+      }
+    } catch (error) {
+      console.log(error);
```
Comment (ncutixavier) [inline src/controllers/userController.js:36]: Return appropriate message.

File: src/middlewares/authenticate.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import jwt from "jsonwebtoken";
+
+export const authenticate = async (req, res, next) => {
+  try {
+    const token = req.headers.authorization.split(" ")[1];
+    await jwt.verify(token, process.env.JWT_SECRETE);
+    next();
+  } catch (error) {
+    return res.status(401).json({
+      status: 401,
```
Comment (ncutixavier) [inline src/middlewares/authenticate.js:11]: Yo don't have to return status again

File: src/validations/userValidation/user.schema.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  email: joi.string().email().required(),
+  password: joi
+    .string()
+    .min(8)
+    .max(10)
+    .pattern(
+      new RegExp(/^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{6,16}$/)
+    )
+    .message("Password must atleast have special character and a number")
+    .required(),
```
Comment (ncutixavier) [inline src/validations/userValidation/user.schema.js:14]: Add custom message

**Review by ncutixavier — CHANGES_REQUESTED** [review]

**Review by Eli250 — COMMENTED** [review]
I have resolved all comments.

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**